### ISSUE TITLE
[DataObject][Preview] Add option to verify in controller if request comes from object preview

### ISSUE THREAD

**Issue by fashxp** [pimcore/pimcore#5164]
- check if request comes from object preview
- check for existing admin session etc. to verify if request 

then also update demo, see https://github.com/pimcore/demo/issues/154

**Comment by dpfaffenbauer** [comment]
basically, everything for that already exists, doesn't it? so, if `$requestHelper->isFrontendRequestByAdmin($request))` let you know that it is a backend admin request, so basically, if it's a fronted controller, it is a preview request. You can also fetch the preview query parameter to be absolutely sure.

**Comment by dpfaffenbauer** [comment]
only issue currently is https://github.com/pimcore/pimcore/issues/5106

**Comment by fashxp** [comment]
but `isFrontendRequestByAdmin` does not check, if really an admin session is there, does it? 
so, if somebody just adds the parameters to the requests, she could simulate a preview request and for example access unpublished data.

**Comment by dpfaffenbauer** [comment]
You can pass the `isFrontendRequestByAdmin` by adding on of these query parameters: 

```
 - 'pimcore_editmode',
 - 'pimcore_preview',
 - 'pimcore_admin',
 - 'pimcore_object_preview',
 - 'pimcore_version
```

But yes, there is a problem, `isFrontendRequestByAdmin` always returns true with one of these parameters set and thus you can see unpublished data under certain circumstances.

**Comment by dvesh3** [comment]
Fixed by #9308
